src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectlyTests.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Usage.CSharpImplementGenericMathInterfacesCorrectly,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Usage.UnitTests
+{
+    public class ImplementGenericMathInterfacesCorrectlyTests
```

Comment:
We should have a test for each interface listed in [https://github.com/dotnet/runtime/issues/69775](https://github.com/dotnet/runtime/issues/69775). One that is missing is `IAdditiveIdentity`

Reply:
I was considering that at first, but 

1. The idea pattern is same and interfaces has similar structure no need to check all 
2. Some of the interface's implemented many other interfaces therefore needs lots of method implementations added

Though I can add few more test with different interfaces

Reply:
I think we should have a test for each. There are a few cases where the names weren't correct and while it is more work, it helps validate things are working as expected end to end.

Reply:
Yeah, I actually noticed some interfaces type parameters are different in the list than in the code, so planning to add more tests